File: students/students/src/MON.Models/Status/ServerInfo.cs
namespace MON.Models.Status
{
    public class ServerInfo
    {
        public string OsPlatform { get; set; }
        public string AspDotnetVersion { get; set; }
        // The memory occupied by objects.
        public long Allocated { get; set; }
        // The value returned by this property represents the current size of memory used by the process, in bytes, that 
        // cannot be shared with other processes.
        public long PrivateMemorySize64 { get; set; }
        public long VirtualMemorySize64 { get; set; }
        // The working set includes both shared and private data. The shared data includes the pages that contain all the 
        // instructions that the process executes, including instructions in the process modules and the system libraries.
        public long WorkingSet64 { get; set; }
        public double CpuUsage { get; set; }
        public int ProcessorCount { get; set; }
        public int MinWorkerThreads { get; set; }
        public int MaxWorkerThreads { get; set; }
        public int AvailableWorkerThreads { get; set; }
        
        public int MinIOThreads { get; set; }
        public int MaxIOThreads { get; set; }
        public int AvailableIOThreads { get; set; }
    }
}

File: students/students/src/MON.SPA/src/models/admin/serverInfo.js
export class ServerInfo{
    constructor(obj = {}){
        this.osPlatform = obj.osPlatform || '';
        this.aspDotnetVersion = obj.aspDotnetVersion || '';
        this.allocated = obj.allocated || 0;
        this.privateMemorySize64 = obj.privateMemorySize64 || 0;
        this.virtualMemorySize64 = obj.virtualMemorySize64 || 0;
        this.workingSet64 = obj.workingSet64 || 0;
        this.cpuUsage = obj.cpuUsage || 0.0;
        this.processorCount = obj.processorCount || 0;
        this.minIOThreads = obj.minIOThreads || 0;
        this.minWorkerThreads = obj.minWorkerThreads || 0;
        this.maxIOThreads = obj.maxIOThreads || 0;
        this.maxWorkerThreads = obj.maxWorkerThreads || 0;
        this.availableIOThreads = obj.availableIOThreads || 0;
        this.availableWorkerThreads = obj.availableWorkerThreads || 0;
    }
}
